desktop_version/src/CustomLevels.cpp

```diff
@@ -1243,6 +1245,59 @@ bool customlevelclass::load(std::string& _path)
                 script.customscripts.push_back(script_);
             }
         }
+
+        if (SDL_strcmp(pKey, "SpecialRoomnames") == 0)
+        {
+            for (tinyxml2::XMLElement* roomnameElement = pElem->FirstChildElement(); roomnameElement; roomnameElement = roomnameElement->NextSiblingElement())
+            {
+                const char* roomnameType = roomnameElement->Value();
+                Roomname name;
+                name.x = 0;
+                name.y = 0;
+                name.flag = -1;
+                name.loop = false;
+                name.type = STATIC;
+                name.progress = 0;
+                name.delay = 0;
+                if (SDL_strcmp(roomnameType, "transform") == 0)
+                {
+                    name.type = TRANSFORM;
+                    name.delay = 2;
+                }
+                else if (SDL_strcmp(roomnameType, "glitch") == 0)
+                {
+                    name.type = GLITCH;
+                    name.progress = 1;
+                    name.delay = -1;
+                }
+
+                name.text.clear();
+
+                roomnameElement->QueryIntAttribute("x", &name.x);
+                roomnameElement->QueryIntAttribute("y", &name.y);
+                roomnameElement->QueryIntAttribute("flag", &name.flag);
+
+                roomnameElement->QueryBoolAttribute("loop", &name.loop);
+
+                if (name.type == STATIC)
+                {
+                    name.text.push_back(std::string(roomnameElement->GetText()));
+                }
+                else
+                {
+                    for (tinyxml2::XMLElement* textElement = roomnameElement->FirstChildElement(); textElement; textElement = textElement->NextSiblingElement())
+                    {
+                        if (SDL_strcmp(textElement->Value(), "text") == 0)
+                        {
+                            name.text.push_back(std::string(textElement->GetText()));
+                        }
+                    }
+                }
+
+                // Push to the start, since the for loop which processes these later `break`s
+                map.specialroomnames.insert(map.specialroomnames.begin(), name);
```

Comment:
Not sure I understand this comment. Why exactly does it matter that this should be pushed to the start of the vector? The for-loop in `updateroomnames` will still have *O*(*n*) runtime on average, this doesn't seem to do anything.

Reply:
I guess this sort of implicit priority is par for the course for this game (e.g. which script box executes first when you touch two script boxes at a time depends on which script box was placed first), but I do find myself thinking there could be something more explicit instead of the order of the tags in the XML. (This is not me saying that it'd be worth it, because I am not sure of that.)

Anyways, regardless of that, it'd be better if you `push_back`ed into the vector as normal, and simply iterated backwards in `updateroomnames`. That way it makes it clearer that it's being handled in reverse in `updateroomnames`.